stdlib/public/Darwin/Accelerate/vImage_Buffer.swift

```diff
@@ -0,0 +1,348 @@
+    /// - Parameter format: A `vImage_CGImageFormat` that describes the source image.
+    /// - Parameter error: Overwritten with the error code if the operation failed.
+    /// - Parameter options: The options to use when performing this operation.
+    ///
+    /// - Returns: An initialized vImage buffer.
+    ///
+    /// This function will instantiate and initialize a vImage buffer from a `CGImage` using a provided `CGImageFormat`.
+    public init?(cgImage: CGImage,
+                 format: vImage_CGImageFormat,
+                 error: inout Int,
+                 flags options: vImage.Options = .noFlags) {
+        
+        self.init()
+        
+        var format = format
+        error = vImageBuffer_InitWithCGImage(&self,
+                                             &format,
+                                             nil,
+                                             cgImage,
+                                             options.flags)
+        
+        if error != kvImageNoError {
+            return nil
+        }
+    }
+    
+    /// Initialize a vImage buffer with the contents of a Core Graphics image,
+    /// using a supplied format.
+    ///
+    /// - Parameter cgImage: A `CGImage` instance to be used as the source.
+    /// - Parameter format: A `vImage_CGImageFormat` that describes the source image/
+    /// - Parameter options: The options to use when performing this operation.
+    ///
+    /// - Returns: An initialized vImage buffer.
+    ///
+    /// This function will instantiate and initialize a vImage buffer from a `CGImage` using a provided `CGImageFormat`.
+    public init?(cgImage: CGImage,
+                 format: vImage_CGImageFormat,
+                 flags options: vImage.Options = .noFlags) {
+        
+        self.init()
+        
+        var format = format
+        let error = vImageBuffer_InitWithCGImage(&self,
+                                                 &format,
+                                                 nil,
+                                                 cgImage,
+                                                 options.flags)
+        
+        if error != kvImageNoError {
+            return nil
+        }
+    }
+    
+    //===----------------------------------------------------------------------===//
+    
+    /// Creates a `CGImage` instance from a vImage buffer, reporting any errors.
+    ///
+    /// - Parameter format: The image format of this vImage buffer.
+    /// - Parameter error: Overwritten with the error code if the operation failed.
+    /// - Parameter options: The options to use when performing this operation.
+    ///
+    /// - Returns: A Core Graphics image containing a representation of the vImage buffer.
+    public func createCGImage(format: vImage_CGImageFormat,
+                              error: inout Int,
+                              flags options: vImage.Options = .noFlags) -> CGImage? {
```

Comment:
This would probably be better as an init on CGImage, but you should discuss that with the CG team.

Reply:
Email sent.

Reply:
Convenience initializers are not supported in extensions of CF types.